compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1091,6 +1091,8 @@ object Parsers {
     /** Singleton    ::=  SimpleRef
      *                 |  SimpleLiteral
      *                 |  Singleton ‘.’ id
+     * -- not yet      |  Singleton ‘(’ Singletons ‘)’
+     * -- not yet      |  Singleton ‘[’ Types ‘]’
```

Comment:
That's not an issue of this PR, but why are `Singletons` required to be non-empty?